transport/rif_comms/node.py

```diff
@@ -1,135 +1,228 @@
+    def _start_message_listener(self):
+        """
+        Start a listener greenlet to listen for received messages in the background.
+        """
+        our_address = to_checksum_address(self.raiden_service.address)
+        self._our_topic_stream = self._comms_client.subscribe_to(our_address)
+        # TODO: remove this after GRPC API request blocking is fixed
+        self._comms_client._get_peer_id(our_address)
+        self._our_topic_thread = spawn(self._receive_messages)
+        self._our_topic_thread.name = f"RIFCommsClient.listen_messages rsk_address:{self.address}"
+
+    def _receive_messages(self):
+        """
+        Iterate over the Notification stream and block thread to receive messages.
+        """
+        for notification in self._our_topic_stream:
+            raiden_message = self._notification_to_message(notification.channelNewData)
+            if raiden_message:
+                self.log.info("incoming message", message=raiden_message)
+                self._handle_message(raiden_message)
+
+    @staticmethod
+    def _notification_to_message(notification_data: ChannelNewData) -> RaidenMessage:
+        """
+        :param notification_data: raw data received by the RIF Comms GRPC API
+        :return: a raiden.Message
+        """
+        content_text = notification_data.data
+        """
+        ChannelNewData has the following structure:
+            from: "16Uiu2HAm8wq7GpkmTDqBxb4eKGfa2Yos79DabTgSXXF4PcHaDhWJ"
+            data: "{\"type\":\"Buffer\",\"data\":[104,101,121]}"
+            nonce: "\216f\225\232d\023e{"
+            channel {
+              channelId: "16Uiu2HAm9otWzXBcFm7WC2Qufp2h1mpRxK1oox289omHTcKgrpRA"
+            }
+        """
+        if content_text:
+            # we first transform the content of the notification data to a dictionary
+            content = json.loads(content_text.decode())
+            # the message is inside the notification data, encoded by the RIF Comms GRPC api
+            message_string = bytes(content["data"]).decode()
+            message_dict = json.loads(message_string)
+            return message_from_dict(message_dict)
+        return None
+
+    def _handle_message(self, message: RaidenMessage):
+        """
+        Handle received Raiden message.
+        """
+        if self.stop_event.ready():
+            return  # ignore when node is stopped
+
+        # process message if its type is expected
+        if isinstance(message, (Delivered, Processed, SignedRetrieableMessage)):
+            self.log.info(
+                "Raiden message received",
+                type=type(message),
+                node=pex(self._raiden_service.address),
+                message=message,
+                sender=pex(message.sender),
+            )
 
-        return f"<{self.__class__.__name__}{node} id:{id(self)}>"
+            try:
+                # acknowledge to sender that their message was received
+                self._ack_message(message)
```

Comment:
i dont think this is ok, a Delivered message has not a response

Thats why we have `_receive_delivered_to_lc` on matrix impl

Reply:
we could take the `if` outside of the `_ack_message` function if you prefer, i'm ok with it either way

Reply:
agree

Reply:
done